README.md

```diff
@@ -1 +1,173 @@
+    }
+}
+```
+
+## Setup Zendesk Webhooks
+
+Here is a helpful [Zendesk post](https://support.zendesk.com/hc/en-us/articles/204890268-Creating-webhooks-with-the-HTTP-target#topic_yf1_fs5_tr) describing the setup of webhooks
+
+### Add a notification target
+
+From [Zendesk Documentation:](https://developer.zendesk.com/rest_api/docs/support/targets)
+
+> Targets are pointers to cloud-based applications and services such as Twitter and Twilio, as well as to HTTP and email addresses. You can use targets with triggers and automations to send a notification to the target when a ticket is created or updated.
+
+Because the we are sending notifying the cloud app of Zendesk triggered events,
+we will need to add an HTTP target which will accept the webhook notifications.
+
+1. Click the Admin icon (sprocket) in the left sidebar
+1. `Settings` > `Extensions`
+1. `Targets tab` > `Add Target`
+1. Select `HTTP` Target
+1. Fill in the following:
+    1. **Title:** Send Mattermost notification when ticket created
+    1. **Url:** `<your_url/zendesk/webhook>`
+    1. **Method:** POST
+    1. **Content Type:** JSON
+
+**Developer Notes:** When testing webhooks locally, you will need to expose your
+localhost:4040 with ngrok
+
+### Add a trigger when new ticket is created
+
+From [Zendesk Documentation:](https://developer.zendesk.com/rest_api/docs/support/triggers)
+
+> A trigger consists of one or more actions performed when a ticket is created or updated. The actions are performed only if certain conditions are met. For example, a trigger can notify the customer when an agent changes the status of a ticket to Solved.
+
+1. Click the Admin icon (sprocket) in the left sidebar
+1. `Settings` > `Business Rules` > `Triggers` > `Add trigger`
+1. Fill in the following:
+    1. **Trigger Name:** Notify Mattermost App when ticket created
+    1. **Description:** When a new ticket is created, trigger a notification and send to Mattermost zendesk cloud app
+    1. **Conditions:**
+        1. Under "Meet ALL of the following conditions"
+        1. `Ticket` `Is` `Created`
+    1. **Actions:**
+        1. In the first select box choose `Notify target`
+        1. In the second select box choose `Send notification when ticket
+           created`
+        1. Past the following in the JSON Body textarea.
+
+```json
+{
+  "title": "{{ticket.title}}",
+  "ticketUrl": "{{ticket.url}}",
+  "ticketDescription": "{{ticket.description}}",
+  "ticketID": "{{ticket.id}}",
+  "ticketPriority": "{{ticket.priority}}",
+  "ticketRequester": "{{ticket.requester.details}}",
+  "ticketType": "{{ticket.ticket_type}}"
+}
+```
+
+## Types
+
+```typescript
+export type AppletContext = {
+  app_id: string;
+  acting_user_id: string;
+  user_id?: string;
+  team_id?: string;
+  channel_id: string;
+  post_id?: string;
+  root_post_id?: string;
+    props?: {[name: string]: string};
+};
+
+export type AppletNotification = {
+  Subject: string;
+  Context: AppletContext;
```

Comment:
Need this change in the proxy plugin

```suggestion
  subject: string;
  context: AppletContext;
```

Reply:
Also Removing.  Not needed in the README.